### ISSUE TITLE
File not created?

### ISSUE THREAD

**Issue by jantman** [5axes/CuraHtmlDoc#5]
I was really sad to see CuraSettingsWriter no longer working, as I've been using and loving it for years (but haven't updated Cura in quite a while). As recommended in https://github.com/5axes/CuraSettingsWriter/issues/19#issuecomment-2090330614 I'm trying this instead.

However, it doesn't seem to be writing any files.

I'm on Cura 5.8.0 on Arch Linux.

I click the sidebar icon and click "Save As". The first weird thing is it pulls up a file dialog that defaults to a `smb://` path instead of a normal local filesystem path. Ignoring that I browse to a directory that I want to save in, enter a file name (`test.html`), and click Save. The dialog closes, but no file is written.

Am I doing something wrong?

**Comment by jantman** [comment]
Using Cura 5.8.0 on Arch Linux, when I try to use this plugin, I see the following in `~/.local/share/cura/stderr.log` when trying to "Save As" and then select `/home/jantman/tmp/foo.html` as the destination file via my file browser:

```
2024-09-22 11:06:05,564 - WARNING - [MainThread] UM.Decorators.deprecated_function [23]: <function createActiveToolProxy at 0x78e77dda49d0> is deprecated (since 5.7.0): UM.ActiveTool is deprecated and will be removed in major SDK release, Use UM.Controller instead
UM/Qt/Bindings/ControllerProxy.py:218: DeprecationWarning: <function createActiveToolProxy at 0x78e77dda49d0> is deprecated (since 5.7.0): UM.ActiveTool is deprecated and will be removed in major SDK release, Use UM.Controller instead
  self.activeToolChanged.emit()
2024-09-22 11:06:05,573 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [448]: file:///home/jantman/.local/share/cura/5.8/plugins/CuraHtmlDoc/CuraHtmlDoc/qml_qt6/CuraHtmlDoc.qml:60: ReferenceError: topRect is not defined
2024-09-22 11:06:15,278 - ERROR - [MainThread] UM.Qt.Bindings.ActiveToolProxy.setProperty [95]: Exception: Unable to set value 'home/jantman/tmp/foo.html' to property 'FileFolder'.
2024-09-22 11:06:15,279 - ERROR - [MainThread] UM.Qt.Bindings.ActiveToolProxy.setProperty [95]: Traceback (most recent call last):
2024-09-22 11:06:15,280 - ERROR - [MainThread] UM.Qt.Bindings.ActiveToolProxy.setProperty [95]:   File "UM/Qt/Bindings/ActiveToolProxy.py", line 93, in setProperty
2024-09-22 11:06:15,281 - ERROR - [MainThread] UM.Qt.Bindings.ActiveToolProxy.setProperty [95]:     option_setter(value)
2024-09-22 11:06:15,281 - ERROR - [MainThread] UM.Qt.Bindings.ActiveToolProxy.setProperty [95]:   File "/home/jantman/.local/share/cura/5.8/plugins/CuraHtmlDoc/CuraHtmlDoc/CuraHtmlDoc.py", line 249, in setFileFolder
2024-09-22 11:06:15,282 - ERROR - [MainThread] UM.Qt.Bindings.ActiveToolProxy.setProperty [95]:     self._openHtmlPage(self._doc_file)
2024-09-22 11:06:15,283 - ERROR - [MainThread] UM.Qt.Bindings.ActiveToolProxy.setProperty [95]:   File "/home/jantman/.local/share/cura/5.8/plugins/CuraHtmlDoc/CuraHtmlDoc/CuraHtmlDoc.py", line 311, in _openHtmlPage
2024-09-22 11:06:15,283 - ERROR - [MainThread] UM.Qt.Bindings.ActiveToolProxy.setProperty [95]:     with open(page_name, 'w', encoding='utf-8') as fhandle:
2024-09-22 11:06:15,284 - ERROR - [MainThread] UM.Qt.Bindings.ActiveToolProxy.setProperty [95]: FileNotFoundError: [Errno 2] No such file or directory: 'home/jantman/tmp/foo.html'
```

It looks like maybe there's a bug on Linux (and maybe Mac too?) where it can't handle POSIX file paths correctly?

**Comment by jantman** [comment]
I've come up with a fix for this in #6
